File: go_files/ex14.go
package main

import (
    "fmt"
    "sync"
)

// Definindo a struct Configuracao
type Configuracao struct {
    config string
}

var instancia *Configuracao
var once sync.Once

func GetInstancia() *Configuracao {
	once.Do(func() {
		instancia = &Configuracao{config: "Configuração padrão"}
	})
	return instancia
}


func main() {
    config1 := GetInstancia()
    config2 := GetInstancia()

    fmt.Println(config1 == config2)  // Saída: true
    fmt.Println(config1.config)      // Saída: Configuração padrão
    config1.config = "Nova configuração"
    fmt.Println(config2.config)      // Saída: Nova configuração
}

File: python_files/ex14.py
class Configuracao:
    _instancia = None

    def __new__(cls):
        if cls._instancia is None:
            cls._instancia = super(Configuracao, cls).__new__(cls)
            cls._instancia.config = "Configuração padrão"
        return cls._instancia
    
# Testando o Singleton
config1 = Configuracao()
config2 = Configuracao()

print(config1 is config2)  # Saída: True
print(config1.config)  # Saída: Configuração padrão
config1.config = "Nova configuração"
print(config2.config)  # Saída: Nova configuração